Wrappers/Python/ccpi/optimisation/operators/FiniteDifferenceOperator.py

```diff
@@ -23,25 +23,29 @@ class FiniteDiff(Operator):
     # Grad_order = ['direction_z', 'direction_y', 'direction_x']
     # Grad_order = ['channels', 'direction_z', 'direction_y', 'direction_x']
     
-    def __init__(self, gm_domain, gm_range=None, direction=0, bnd_cond = 'Neumann'):
+    def __init__(self, gm_domain, gm_range=None, direction=0, bnd_cond = 'Neumann', **kwargs):
         ''''''
         super(FiniteDiff, self).__init__() 
-        '''FIXME: domain and range should be geometries'''
+
         self.gm_domain = gm_domain
         self.gm_range = gm_range
+        
         self.direction = direction
         self.bnd_cond = bnd_cond
-        
+                
+        ''' Domain Geometry = Range Geometry'''
         if self.gm_range is None:
             self.gm_range = self.gm_domain
-            
-        if self.direction + 1 > len(gm_domain):
-            raise ValueError('Gradient directions more than geometry domain')      
-                         
+        
+        ''' check direction and "length" of geometry  '''
+        if self.direction + 1 > len(self.gm_domain):
+            raise ValueError('Gradient directions more than geometry length') 
+
+        self.voxel_size = kwargs.get('voxel_size',1)            
```

Comment:
`voxel_size` should be in the `ImageGeometry` as `voxel_size_x` `voxel_size_y` `voxel_size_z`.
As `pixel_size_h` and `pixel_size_v` in the `AcquisitionGeometry`. 

The point is that these info in the geometries have not been used and are not particularly handy. For instance there is no reference to the unit measure of the number stored in those variables.

Reply:
Your implementation also postulates homogeneous voxel size.